test/fixtures/google-gax-packaging-test-app/src/index.js

```diff
@@ -97,6 +103,56 @@ async function testEcho(client) {
   assert.deepStrictEqual(request.content, response.content);
 }
 
+async function testEchoError(client) {
+  const readFile = util.promisify(fs.readFile);
+
+  const fixtureName = path.resolve(
+    __dirname,
+    '..',
+    '..',
+    '..',
+    'test',
+    'fixtures',
+    'multipleErrors.json'
+  );
+  const protos_path = path.resolve(
+    __dirname,
+    '..',
+    '..',
+    '..',
+    'protos',
+    'google',
+    'rpc'
+  );
+
+  const data = await readFile(fixtureName, 'utf8');
+  const root = protobuf.loadSync(
+    path.join(protos_path, 'error_details.proto')
+  );
+  const objs = JSON.parse(data);
+  for (const obj of objs) {
+    const MessageType = root.lookupType(obj.type);
+    const buffer = MessageType.encode(obj.value).finish();
+    const request = {
+      error: {
+        code: 3,
+        message: 'Test error',
+        details: [{
+          type_url: 'type.googleapis.com/' + obj.type,
+          value: buffer,
+        }],
+      },
+    };
+    const timer = setTimeout(() => {
+      throw new Error('End-to-end testEchoError method fails with timeout');
+    }, 12000);
+    await assert.rejects(() => client.echo(request),
+    Error);
```

Comment:
Good. But we also need to verify the error field if they contains the statusDetails, right?

Reply:
try-catch was added back to the integration test to check the new field statusDetail